### PR TITLE
Remove unnecessary servlet response manipulations

### PR DESCRIPTION
**Pull request by dsyer**

### CODE REVIEW

**Comment by agrison** [comment]
I don't see why this is a very big problem (related to _uglyness_) for the BE to drive the URL for these pagination steps.

With the changes you propose, the URL are no more bookmarkable when changing page, the URL doesn't get updated anymore and stays on `/owners/find` instead of say `/owners/find?lastName=Davis&page=1`.

We would need to fix that, also I think on `vets.html` but this was a missing feature probably, it should have worked the same way it worked on searching owners.

**Comment by dsyer** [comment]
You don't think servlet API references in controllers is ugly? I suppose it's not for me to say.

Anyway, I see what you mean about the URL query params being bookmarkable - I don't recall ever seeing them, but I was probably not looking hard enough. The vets link has no query params so I might hack the PR around until it is the smallest it can be without changing bookmarks.

**Comment by dsyer** [comment]
I updated the `OwnerController` to preserve the form query params in the `hx-push-url` using `HtmxResponse` from the HTMX library. The other endpoints all seem fine without changing the controllers, or maybe I missed something.

File: src/main/resources/templates/fragments/owners.html [hunk 1]
```diff
@@ -122,9 +122,13 @@ <h2>Pets and Visits</h2>
           </tr>
           <tr>
             <td><a th:href="@{__${owner.id}__/pets/__${pet.id}__/edit}"
-                   th:attr="hx-get=@{__${owner.id}__/pets/__${pet.id}__/edit}" hx-target="#block-content">Edit Pet</a></td>
+                   hx:get="@{__${owner.id}__/pets/__${pet.id}__/edit}"
+                   hx:push-url="@{__${owner.id}__/pets/__${pet.id}__/edit}"
```
Comment (wimdeblauwe) [inline src/main/resources/templates/fragments/owners.html]: You could replace this with `hx-push-url="true"` and it should work just the same.

Comment (dsyer) [reply src/main/resources/templates/fragments/owners.html]: Done. I also replaced all the unnecessary `HttpServletRequest` and `HttpServletResponse` method parameters. I wonder if we even need the `HtmxResponse` (controller methods can return strings if there is only one view to render), but I didn't go that far.

**Pull request merged**